### ISSUE TITLE
Tmux-New-{Vertical,Horizontal} Incorrect Description

### ISSUE THREAD

**Issue by bhajneet** [mawww/kakoune#1626]
Current descriptions:

```
tmux-new-vertical
Create a new vertical pane

tmux-new-horizontal
Create a new horizontal pane
```

This isn't what actually happens. Instead what happens is closer to this description:

```
tmux-new-vertical
Vertically split to create a new horizontal pane

tmux-new-horizontal
Horizontally split to create a new vertical pane
```

I know the tmux command is split-window -v to create a horizontal pane. In my opinion that is confusing in tmux as well.

It should be clear whether we are splitting in a direction or describing the new pane's orientation that is created. Currently the description is the latter but does the former.

Perhaps new commands?

```
tmux-new-vertical
split-window -h

tmux-new-horizontal
split-window -v

tmux-split-vertically
split-window -h

tmux-split-horizontally
split-window -v
```

**Comment by Delapouite** [comment]
I'm also always confused in any system that use splits (i3, xmonad, tmux like here, vim…)

I never remember if "split this thing horizontally" means that they two resulting parts are `left` | `right` or that the "split separator" is horizontal and therefore the two resulting parts are `up` / `down`.

**Comment by lenormf** [comment]
Maybe the docstrings could be fixed to used the same wording used in the `tmux` documentation:

```
%           Split the current pane into two, left and right.
"           Split the current pane into two, top and bottom.
```

**Comment by mawww** [comment]
I agree we should make it clear what we consider vertical and horizontal splitting (using a better wording, such as the one tmux uses). Then ensure that our splitting implementation do the same (regardless of what the underlying implementation considers vertical/horizontal).

Or should we go the other way around ? And let the actual implementation decide what vertical/horizontal means ? That would make things easier for interactive use, as users do know if they are running inside tmux or i3 or whatever and hence should know what horizontal/vertical means in that context. On the other hand, it makes it harder for scripts to predict what is going to happen, for example a potential diff script will really want to display buffers side by side.

So, now I am confused...

**Comment by bhajneet** [comment]
It's best to keep the commands and their descriptions clear. The only downside is that previous scripts using the old functionality will be broken and need updating. Because either way we have the wrong description or the wrong command names. The clearest way to define the commands is using the tmux pane as a subject, and the verb splitting or creating (new implies creation) to describe the action.

```
tmux-split-pane-vertically   => tmux-split-vertically (correctly named command for old implementation)
tmux-split-pane-horizontally => tmux-split-horizontally
tmux-new-vertical-pane       => tmux-new-vertical (change functionality to match description)
tmux-new-horizontal-pane     => tmux-new-horizontal

```

Using tmux documentation would be a bad idea. The command `"` is `split-pane -v`. When the words itself say "split pane" and it adds a "v" for vertical, you may think "split the pane vertically" and not "split the pane to *add a* vertical" as it actually does.

The problem is in the nuance of vertical vs vertically. Usually you split things in a direction. So I would argue tmux is also incorrect. You don't split something and then say "look what I added, a vertical pane". You usually "cut it" in a direction.

I hope this clears things up.

**Comment by bhajneet** [comment]
Or put very simply:

Split = divide

New = add

You can divide a pane either vertically or horizontally (the result being a new pane in the opposite direction)

And you can add a pane either vertical or horizontal (the result being the current pane is cut in the opposite direction)

**Comment by robertmeta** [comment]
I think the way Emacs describes splitting is very clear.

C-x 2
    Split the selected window into two windows, one above the other (split-window-below).
C-x 3
    Split the selected window into two windows, positioned side by side (split-window-right).

**Comment by mawww** [comment]
I am happy for all that to be cleaned up, I think if we change the tmux command names, we should probably tweak the other ones (screen, iterm2, ...) and unify on a naming scheme.
